### DISCUSSION TITLE
Vite/Ionic

### DISCUSSION THREAD

**Discussion by ZajdeR** [setaman/vue-ellipse-progress#235] [General]
Is this library working with Vite?

**Comment by setaman** [comment]
Sure! using it right now in several Vite powered projects, without any issues

**Comment by ZajdeR** [comment]
It caused me an error:
![image](https://github.com/setaman/vue-ellipse-progress/assets/13215441/50d132ca-d90a-41de-a234-8df6e101e5e5)

I'm installing it according to documentation.

**Reply by setaman** [reply]
could you please share more details from your set up, how you are installing the plugin, using and configuring it, also supplied props etc.

**Reply by ZajdeR** [reply]
Sure:
node: v18.12.1
```
dependencies": {
    "@ionic/vue": "^7.6.6",
    "@ionic/vue-router": "^7.6.6",
    "pinia": "^2.1.7",
    "vue": "^3.3.11",
    "vue-ellipse-progress": "^2.1.2",
    "vue-i18n": "^9.9.0",
    "vue-router": "^4.2.5"
  },
```
Error occurs just when I add in `main.js`:
`import veProgress from 'vue-ellipse-progress';`
and project rebuilds. 
Even when I remove import, error stil exists. I have to remove dependency and then it works again.

**Reply by setaman** [reply]
Thank you! 

I have no experience working with Ionic. Following the official [docs](https://ionicframework.com/docs/vue/overview#installation) i bootstrapped a minimal app and anything seems to work like expected, at least in the browser
![image](https://github.com/setaman/vue-ellipse-progress/assets/15034996/fe2c086e-0990-4fbc-962b-aa0d72468a9e)

Is there something different/special in your project?

**Reply by ZajdeR** [reply]
Are you importing vue-router package? It seems that this causing error in my project.

**Reply by setaman** [reply]
yes, router is also included in the default `tab` preset i used

**Reply by ZajdeR** [reply]
It finally works :roll_eyes: when I put the library import before the ionic/vue import it started working.

**Reply by setaman** [reply]
oh, that's strange, but happy it works now!
